src/err/err_state.rs

```diff
@@ -68,11 +68,17 @@ impl PyErrState {
                     )
                 }
             }
-            PyErrState::LazyValue { ptype, pvalue } => (
-                ptype.into_ptr(),
-                pvalue(py).into_ptr(),
-                std::ptr::null_mut(),
-            ),
+            PyErrState::LazyValue { ptype, pvalue } => {
+                if unsafe { ffi::PyExceptionClass_Check(ptype.as_ptr()) } == 0 {
+                    Self::exceptions_must_derive_from_base_exception(py).into_ffi_tuple(py)
+                } else {
```

Comment:
I moved this check from `PyErr::from_type` to here. This allowed me to observe the case of normalization failure easier in the test (by delaying this check until normalisation). It's also now more consistent with the match arm above, which also does this check.